src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java

```diff
@@ -318,19 +319,38 @@ public void registerAttribute(@NotNull Attribute attribute)
 	@Override
 	public <T extends Projectile> @NotNull T launchProjectile(@NotNull Class<? extends T> projectile)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return this.launchProjectile(projectile, null);
 	}
 
 	@Override
 	public <T extends Projectile> @NotNull T launchProjectile(@NotNull Class<? extends T> projectile, @Nullable Vector velocity)
+	{
+		return this.launchProjectile(projectile, velocity, null);
+	}
+
+	@Override
+	public <T extends Projectile> @NotNull T launchProjectile(@NotNull Class<? extends T> projectile, @Nullable Vector velocity, @Nullable Consumer<? super T> function)
 	{
 		Preconditions.checkNotNull(projectile, "Projectile cannot be null");
-		T entity = launchProjectile(projectile);
+
+		World world = getWorld();
+		Preconditions.checkNotNull(world, "World cannot be null when launching a projectile");
+
+		// The throwable spawn location is slightly lower than the eye level (See: ThrowableProjectile constructor)
+		Location spawnLocation = getEyeLocation().subtract(0, 0.10000000149011612D, 0);
+		T entity = world.createEntity(spawnLocation, projectile);
```

Comment:
Just to be sure here, can every class that extends `Projectile` be spawned in the world using createEntity?

Reply:
I will update the unit tests to include this, but some of them are not implemented yet:
![image](https://github.com/user-attachments/assets/4926f7a5-6203-48ff-b9c5-af051756f7f5)

Reply:
Ah, that looks alright, I was mostly wondering if the "abstract" classes were included, like `AbstractArrow`, which should redirect another class.

Anyhow, that is for a separate pr if that's an issue anyhow.